[buildbot] Adopt webkitcorepy's test harness

JonWBedard opened the pull request:
#### 1eac3c959b1ce039b2e6be27bc3b77f92fa0c96a
<pre>
[buildbot] Adopt webkitcorepy&apos;s test harness
<a href="https://bugs.webkit.org/show_bug.cgi?id=258526">https://bugs.webkit.org/show_bug.cgi?id=258526</a>
rdar://111339627

Reviewed by Aakash Jain.

* Tools/CISupport/build-webkit-org/__init__.py: Added.
* Tools/CISupport/build-webkit-org/factories.py: Import from Tools/CISupport as the root.
* Tools/CISupport/build-webkit-org/factories_unittest.py: Ditto.
* Tools/CISupport/build-webkit-org/loadConfig.py:
(loadBuilderConfig): Use explicit path to config.json.
* Tools/CISupport/build-webkit-org/loadConfig_unittest.py: Import from Tools/CISupport as the root.
* Tools/CISupport/build-webkit-org/master.cfg: Ditto.
* Tools/CISupport/build-webkit-org/steps_unittest.py: Ditto.
* Tools/CISupport/build-webkit-org/wkbuild_unittest.py: Ditto.
* Tools/CISupport/ews-build/__init__.py: Added.
* Tools/CISupport/ews-build/events.py: Import from Tools/CISupport as the root.
* Tools/CISupport/ews-build/events_unittest.py: Ditto.
* Tools/CISupport/ews-build/factories.py: Ditto.
* Tools/CISupport/ews-build/factories_unittest.py: Ditto.
* Tools/CISupport/ews-build/loadConfig.py: Ditto.
* Tools/CISupport/ews-build/loadConfig_unittest.py: Ditto.
* Tools/CISupport/ews-build/master.cfg: Ditto.
* Tools/CISupport/ews-build/results_db.py: Ditto.
* Tools/CISupport/ews-build/steps.py: Ditto.
* Tools/CISupport/ews-build/steps_unittest.py: Ditto.
* Tools/CISupport/run-tests: Added.
* Tools/CISupport/runUnittests.py:
(run_unittests): Import from Tools/CISupport as the root.
* Tools/Scripts/webkitpy/autoinstalled/buildbot.py: Update Tempita, add lz4.

Canonical link: <a href="https://commits.webkit.org/267005@main">https://commits.webkit.org/267005@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ab055489ef030909cec698683531d01ba44d5f67

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/15342 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/15645 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/16007 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/17098 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/14401 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/15485 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/18162 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/15745 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/17000 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/15525 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/15952 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/13048 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/17833 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/15447 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/13230 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/13843 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/20790 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/14312 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/14010 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/17266 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/14583 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/12359 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/15240 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/13853 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/18200 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/1864 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/14416 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aj062 reviewed (approved):
rs=me

Error messages can be improved when the arguments to the scripts are invalid.

JonWBedard committed:
[buildbot] Adopt webkitcorepy's test harness
https://bugs.webkit.org/show_bug.cgi?id=258526
rdar://111339627

Reviewed by Aakash Jain.

* Tools/CISupport/build-webkit-org/__init__.py: Added.
* Tools/CISupport/build-webkit-org/factories.py: Import from Tools/CISupport as the root.
* Tools/CISupport/build-webkit-org/factories_unittest.py: Ditto.
* Tools/CISupport/build-webkit-org/loadConfig.py:
(loadBuilderConfig): Use explicit path to config.json.
* Tools/CISupport/build-webkit-org/loadConfig_unittest.py: Import from Tools/CISupport as the root.
* Tools/CISupport/build-webkit-org/master.cfg: Ditto.
* Tools/CISupport/build-webkit-org/steps_unittest.py: Ditto.
* Tools/CISupport/build-webkit-org/wkbuild_unittest.py: Ditto.
* Tools/CISupport/ews-build/__init__.py: Added.
* Tools/CISupport/ews-build/events.py: Import from Tools/CISupport as the root.
* Tools/CISupport/ews-build/events_unittest.py: Ditto.
* Tools/CISupport/ews-build/factories.py: Ditto.
* Tools/CISupport/ews-build/factories_unittest.py: Ditto.
* Tools/CISupport/ews-build/loadConfig.py: Ditto.
* Tools/CISupport/ews-build/loadConfig_unittest.py: Ditto.
* Tools/CISupport/ews-build/master.cfg: Ditto.
* Tools/CISupport/ews-build/results_db.py: Ditto.
* Tools/CISupport/ews-build/steps.py: Ditto.
* Tools/CISupport/ews-build/steps_unittest.py: Ditto.
* Tools/CISupport/run-tests: Added.
* Tools/CISupport/runUnittests.py:
(run_unittests): Import from Tools/CISupport as the root.
* Tools/Scripts/webkitpy/autoinstalled/buildbot.py: Update Tempita, add lz4.

Canonical link: https://commits.webkit.org/267005@main
